python/graphstorm/inference/np_infer.py

```diff
@@ -168,3 +171,90 @@ def infer(self, loader, save_embed_path, save_prediction_path=None,
             save_node_prediction_results(shuffled_preds, save_prediction_path)
         barrier()
         sys_tracker.check('save predictions')
+
+
+class GSGnnNodePredictionRealtimeInferrer(GSInferrer):
+    """ Inferrer for real-time node prediction tasks on SageMaker endpoints.
+
+    .. versionadded:: 0.5
+        Add `GSGnnNodePredictionRealtimeInferrer` class in v0.5 to support real-time node-level
+        inference on Amazon SageMaker endpoints.
+
+    The real-time inferrer has three major differences from the node prediction offline inferrer:
+
+    1. directly setting and using a DGL graph as input to extract features.
+    2. using DGL dataloader, instead of GS dataloaders that rely on DGL distributed graphs.
+    3. no evaluation section as there is no label on real-time inference.
+    """
+    def infer(self,
+              g,
+              dataloader,
+              infer_ntypes,
+              nfeat_fields,
+              efeat_fields=None,
+              return_proba=True):
+        """
+        ``GSGnnNodePredictionRealtimeInferrer`` defines the ``infer()`` method that performs
+        three tasks:
+
+        1. Extract one batch using the given dataloader;
+        2. Prepare input node and edge features;
+        3. Compute inference results for nodes with target node type and return the results.
+
+        Parameters
+        ----------
+        g: DGLGraph
+            The inference graph data in the format of a DGL heterograph. For built-in inference
+            pipeline, this graph should be constructed by using methods in the
+            `gconstruct.construct_payload_graph.py` file.
+        dataloader: DGL dataloader
+            A DGL dataloader class for node prediction, e.g., `DataLoader`.
+        infer_ntypes: list of string or a string
```

Comment:
Can we make it a list of strings?

Reply:
The format of a string and a list of string is compatible with GS's `target_ntype` argument. I would suggest to keep this compatibility.